### DISCUSSION TITLE
Deploying to production

### DISCUSSION THREAD

**Discussion by zekena2** [awslabs/soci-snapshotter#414] [Q&A] [answered]
Hi, I've some question to clarify some things because the docs aren't answering everything.
Is the project production ready?
Does ECR support the Referers API? Can I use it with soci-snapshotter instead of ORAS that's mentioned in the getting started guide?
Can I use docker command for buidling and tagging images and then creating the index and pushing them to the registry?

Thanks in advance.

**Comment by Kern--** [comment, marked as answer]
Hi, thanks for your interest in SOCI!

> Is the project production ready?

We've spent quite a bit of effort over the last couple of months to get to production readiness. We're pretty confident in it's current state, but we haven't quite thrown enough varied workloads and conditions at it to call it production ready just yet. I would place it in the beta phase.

> Does ECR support the Referers API? Can I use it with soci-snapshotter instead of ORAS that's mentioned in the getting started guide?

ECR does not support the referrers API yet, but we recently added support for existing registries without requiring the referrers API via `soci create --legacy-registry $IMAGE`. This will work with ECR. This flag is pretty poorly named and we're planning to remove the flag and make it default soon https://github.com/awslabs/soci-snapshotter/issues/434. 

> Can I use docker command for buidling and tagging images and then creating the index and pushing them to the registry?

Our tooling expects to be able to get image data from containerd. There is an experimental feature in newer versions of docker that use containerd for image management: https://www.docker.com/blog/extending-docker-integration-with-containerd/. The details are sparse and I haven't tried it myself, but that should allow you to build and tag with docker.

Alternatively [nerdctl](https://github.com/containerd/nerdctl) offers a docker-like CLI experience but directly interacts with containerd.

**Comment by zekena2** [comment]
Thanks for your answer, good to know that ECR is supported I thought it'd be a blocker.
